helper.go

```diff
@@ -669,6 +677,9 @@ func parseRawField(logger Logger, data []byte, msg string) (interface{}, int, er
 		if length > len(data) {
 			return nil, 0, fmt.Errorf("not enough data for Integer (%d vs %d): %x", length, len(data), data)
 		}
+		if cursor > length {
```

Comment:
Here and similar code that follows: Avoid a panic due to referencing `data[cursor:length]` when `cursor` is a higher value than `length`.